packages/babel-jest/src/index.js

```diff
@@ -18,75 +18,34 @@ import type {
 import crypto from 'crypto';
 import fs from 'fs';
 import path from 'path';
+import {
+  transform as babelTransform,
+  util as babelUtil,
+  loadPartialConfig,
+} from '@babel/core';
 import jestPreset from 'babel-preset-jest';
-import {transform as babelTransform, util as babelUtil} from 'babel-core';
 import babelIstanbulPlugin from 'babel-plugin-istanbul';
 
-const BABELRC_FILENAME = '.babelrc';
-const BABELRC_JS_FILENAME = '.babelrc.js';
-const BABEL_CONFIG_JS_FILENAME = 'babel.config.js';
-const BABEL_CONFIG_KEY = 'babel';
-const PACKAGE_JSON = 'package.json';
 const THIS_FILE = fs.readFileSync(__filename);
 
 const createTransformer = (options: any): Transformer => {
-  const cache = Object.create(null);
-
-  const getBabelRC = filename => {
-    const paths = [];
-    let directory = filename;
-    while (directory !== (directory = path.dirname(directory))) {
-      if (cache[directory]) {
-        break;
-      }
-
-      paths.push(directory);
-      const configFilePath = path.join(directory, BABELRC_FILENAME);
-      if (fs.existsSync(configFilePath)) {
-        cache[directory] = fs.readFileSync(configFilePath, 'utf8');
-        break;
-      }
-      let configJsFilePath = path.join(directory, BABELRC_JS_FILENAME);
-      if (fs.existsSync(configJsFilePath)) {
-        // $FlowFixMe
-        cache[directory] = JSON.stringify(require(configJsFilePath));
-        break;
-      }
-      configJsFilePath = path.join(directory, BABEL_CONFIG_JS_FILENAME);
-      if (fs.existsSync(configJsFilePath)) {
-        // $FlowFixMe
-        cache[directory] = JSON.stringify(require(configJsFilePath));
-        break;
-      }
-      const resolvedJsonFilePath = path.join(directory, PACKAGE_JSON);
-      const packageJsonFilePath =
-        resolvedJsonFilePath === PACKAGE_JSON
-          ? path.resolve(directory, PACKAGE_JSON)
-          : resolvedJsonFilePath;
-      if (fs.existsSync(packageJsonFilePath)) {
-        // $FlowFixMe
-        const packageJsonFileContents = require(packageJsonFilePath);
-        if (packageJsonFileContents[BABEL_CONFIG_KEY]) {
-          cache[directory] = JSON.stringify(
-            packageJsonFileContents[BABEL_CONFIG_KEY],
-          );
-          break;
-        }
-      }
-    }
-    paths.forEach(directoryPath => (cache[directoryPath] = cache[directory]));
-    return cache[directory] || '';
-  };
-
   options = Object.assign({}, options, {
     compact: false,
     plugins: (options && options.plugins) || [],
     presets: ((options && options.presets) || []).concat([jestPreset]),
     sourceMaps: 'both',
   });
+
   delete options.cacheDirectory;
   delete options.filename;
 
+  const loadBabelOptions = filename =>
```

Comment:
@SimenB Updated to use `loadPartialConfig`.

Reply:
please add `supportsStaticESM: false` as well